### ISSUE TITLE
[Bug] Pairdrop user gets offline while still choosing files to send

### ISSUE THREAD

**Issue by svl168** [schlagmichdoch/PairDrop#247]
Pairdrop user gets offline while still choosing files to send or still taking picture to send.

**To Reproduce**
1. 2 users (ex. Adam and Eve) using each other's mobile data and running pairdrop at the same time in a chrome browser of their android phone.  Both have previously been paired. and names were permanently named already
2. Adam saw Eve in his pairdrop and clicked her icon in an attempt to send a picture in his gallery
3. Few seconds have passed like even just 3 seconds after, Eve noticed Adam disappeared already like offline.
4. After Adam has chosen a picture to send and clicked send or OK, Eve disappeared already. so, have to repeat again and have to be very fast this time around to avoid getting offline.
5. This does not happen if you're in same WIFI network or at least I did not notice it happen yet.

**Expected behavior**
online status should stay online (each other's icon) while one of them is still choosing files to send or taking pictures.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: Samsung S23 Ultra
 - OS: android 14 one ui 6
 - Browser chrome or brave
 - Version latest

**Bug occurs on official PairDrop instance https://pairdrop.net/**
Yes
Version: v1.10.5

**Bug occurs on self-hosted PairDrop instance**
No | Yes

**Additional context**

**Comment by schlagmichdoch** [comment]
The server checks periodically if a client is still online. The problem might be that the JavaScript code in the browser is stopped by some browsers when their file selector is opened.

> online status should stay online (each other's icon) while one of them is still choosing files to send or taking pictures.

Maybe it'd be more stable if the server stopped as long as a file selector is open. Maybe it would also be sufficient to increase the timeout period.

Is the issue consistently reproducible?

**Comment by svl168** [comment]
Yes, reproducible all the time.  Take note, occurs too while taking a picture.  Thanks for responding.  Other than that, your program is just perfect.

**Comment by schlagmichdoch** [comment]
As [the repository was falsely flagged and hidden for the last month](https://github.com/schlagmichdoch/PairDrop/issues/284) this probably never reached you so this is a friendly bump.

@svl168  I would appreciate it, if you could test this feature on the instance above.

**Comment by ngerberQuad** [comment]
Hi, I'm not the original poster, but I noticed the same bug on my self hosted server.
I tested the it on the development instance and the timeout did not happen again.

**Comment by schlagmichdoch** [comment]
Great! Thanks a lot for testing! I'll hurry up to get this released as part of [v2](#267) soon. I'll keep you posted

**Comment by fm-sys** [comment]
Hi @schlagmichdoch, can you estimate when the fix will get rolled out? Would be nice if we can finally get the app back working...

**Comment by Splitpeach** [comment]
This is still an issue. Also an issue using apps on both devices not just via the browser. Looks like this has been a problem for well over a year now.

**Comment by Splitpeach** [comment]
Im using Linux Mint on laptop and CalyxOS on mobile device btw.

**Comment by gingerbbm** [comment]
Pairdrop is great but by the time I've selected the media on my phone to share with my desktop computer, the connection is lost. I believe that this is the same issue described in this ticket. I'm using a Pixel 7a as the source, and Chrome on a MacBook as the recipient.

**Comment by Splitpeach** [comment]
Still an issue? Is this ever going to get fixed? Been reported well over a year ago now.

**Comment by awolfie** [comment]
This is still an issue for me.  The work around is okay on android play store version, to pick the file(s) and then click share to Pairdrop.  But if I go to Pairdrop and then try to pick the files I loose connection if I am not fast enough.

Otherwise, it is a great app.

**Comment by Payperface** [comment]
Yep, everyone using this in my house is having this issue. Cant really use the site because of this. Such a shame. Would be great to have a fix.

**Comment by schlagmichdoch** [comment]
Please be patient. I’ll have time to patch this in about 1 month

**Comment by naijaping** [comment]
+1

**Comment by gzk-x64** [comment]
+1 🫣

**Comment by SaturnusDJ** [comment]
@schlagmichdoch Is this issue still planned to be resolved? Asking because the last updates are 1,5 years ago, so on the outside it looks like this project is dying. 😬

As an alternative now using (self-hosted) https://github.com/mat-sz/filedrop but that seems to often not do well when the amount of files is above 10. LocalSend only works on LAN and AlterSend requires a license.
